mobile-save-for-later/conf/cfn.yaml

```diff
@@ -34,11 +13,78 @@ Parameters:
     AllowedValues:
       - CODE
       - PROD
+
 Mappings:
   StageVariables:
     CODE:
-      TableReadCapacity: 1
-      TableWriteCapacity: 1
+      TableReadMinCapacity: 1
+      TableReadMaxCapacity: 10
+      TableWriteMinCapacity: 1
+      TableWriteMaxCapacity: 10
     PROD:
-      TableReadCapacity: 200
-      TableWriteCapacity: 125
+      TableReadMinCapacity: 50
+      TableReadMaxCapacity: 400
+      TableWriteMinCapacity: 50
+      TableWriteMaxCapacity: 400
+
+Resources:
+  SaveForLaterDynamoTable:
+    DeletionPolicy: Retain
+    Type: 'AWS::DynamoDB::Table'
+    Properties:
+      KeySchema:
+        - KeyType: HASH
+          AttributeName: userId
+      TableName: !Sub '${App}-${Stage}-articles'
+      AttributeDefinitions:
+        - AttributeName: userId
+          AttributeType: S
+      ProvisionedThroughput:
+        WriteCapacityUnits: !FindInMap [ StageVariables, !Ref Stage, TableWriteMinCapacity ] # Initial value, autoscaled with below resources
+        ReadCapacityUnits: !FindInMap [ StageVariables, !Ref Stage, TableReadMinCapacity ] # Initial value, autoscaled with below resources
+
+  SaveForLaterWritesScalableTarget:
+    Type: AWS::ApplicationAutoScaling::ScalableTarget
+    Properties:
+      MaxCapacity: !FindInMap [StageVariables, !Ref Stage, TableWriteMaxCapacity]
+      MinCapacity: !FindInMap [StageVariables, !Ref Stage, TableWriteMinCapacity]
+      ResourceId: !Sub "table/${App}-${Stage}-articles"
+      RoleARN: !Sub arn:aws:iam::${AWS::AccountId}:role/aws-service-role/dynamodb.application-autoscaling.amazonaws.com/AWSServiceRoleForApplicationAutoScaling_DynamoDBTable
+      ScalableDimension: dynamodb:table:WriteCapacityUnits
+      ServiceNamespace: dynamodb
+
+  SaveForLaterWritesScalingPolicy:
+    Type: AWS::ApplicationAutoScaling::ScalingPolicy
+    Properties:
+      PolicyName: SaveForLaterWritesScalingPolicy
+      PolicyType: TargetTrackingScaling
+      ScalingTargetId: !Ref SaveForLaterWritesScalableTarget
+      TargetTrackingScalingPolicyConfiguration:
+        TargetValue: 70.0
```

Comment:
Might we want a larger target value for the writes compared to the reads? I may be wrong, but I think this one is has a higher usage than reads

Reply:
Oh, I see this is a percentage rather than capacity units

Reply:
My understanding is that this percentage represents the threshold at which AWS will start to scale out the read/write capacity if this percentage utilisation is breached. The minimum/maximum capacity values are those set at the top of the file. Whilst the read/write demands are indeed slightly different, the historical peaks seem to be for reads rather than writes, whereas the daily demand is for higher read capacity. I'm satisfied that using the same min/max values is a good enough starting point.

Reply:
Yeah makes sense